### PR TITLE
Support providing k8s native resource through --objects flag in kanctl

### PR DESCRIPTION
**Pull request by viveksinghggits**
## Change Overview
Support providing k8s native resource through --objects flag in kanctl command, while creating
actionset.
For ex
```
kanctl create actionset \
--action backup \
--namespace kanister \
--blueprint etcd-blueprint \
--objects /v1/pods/kube-system/etcd-minikube \
--profile kube-system/s3-profile-xhgl9
```
Per current implementation we are not able to do this. The validation fails.

Edit: 
we have changed the format and now the core group's resources should be provided using the format 
```
--objects v1/resource/namespace/name
```
for example 
```
--object v1/pods/kube-system/etcd-minikube 
```
And we dont support the format `/v1/resource/namespace/name` for the resources of core api group.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [x] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan
Provided on CR, pod and deployment through --objects flag while creating the actionset.

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by PrasadG193** [comment]
minor: does it work without leading `/` in object resource path 
i.e `--objects v1/pods/kube-system/etcd-minikube`
`

**Comment by viveksinghggits** [comment]
Hey @PrasadG193 
No it doesnt, it doesnt. We are eventually mapping gvr with this value, having this format will work in the cases when we have to actually provide group as well.

**Comment by PrasadG193** [comment]
LG. I'll let @tdmanv to decide whether we want to support format like this `--objects v1/pods/kube-system/etcd-minikube`

**Review by pavannd1 — APPROVED** [review]
LGTM. Any test plan?

**Comment by viveksinghggits** [comment]
@pavannd1 I thought the same but I didn't see any test file, I can add.

**Comment by viveksinghggits** [comment]
> LGTM. Any test plan?

I am adding the test.

File: pkg/kanctl/actionset_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+				Namespace:  "kanister",
+			},
+			err: IsNil,
+		},
+	} {
+		a, err := parseGenericObjectReference(tc.objectFlag)
+		c.Check(err, tc.err)
+		c.Assert(a, DeepEquals, tc.expected)
+	}
+
```
Comment (pavannd1) [inline pkg/kanctl/actionset_test.go:78]:
```suggestion
```

**Review by pavannd1 — APPROVED** [review]
LGTM

**Comment by viveksinghggits** [comment]
Thanks @pavannd1

**Pull request merged**